scripts/lib/CIME/case.py

```diff
@@ -617,13 +620,19 @@ def _get_component_config_data(self, files):
             comp_config_file = self.get_resolved_value(comp_config_file)
             expect(comp_config_file is not None and os.path.isfile(comp_config_file),
                    "Config file {} for component {} not found.".format(comp_config_file, comp_name))
-            compobj = Component(comp_config_file)
-            self._component_description[comp_class] = compobj.get_description(self._compsetname)
-            expect(self._component_description[comp_class] is not None,"No description found in file {} for component {}".format(comp_config_file, comp_name))
+            compobj = Component(comp_config_file, comp_class)
+            compsetname = self._compsetname
+            if not compsetname.endswith('ESP'):
```

Comment:
I've pushed an alternative method to deal with it.   It leaves a potential for errors in the esp component but is otherwise more robust.